### PR TITLE
Enforce Collateralization (for real)

### PR DESCRIPTION
**Pull request by saturnial**

### CODE REVIEW

File: __test__/unit/modules/RequestLoan/RequestLoanForm/RequestLoanForm.test.tsx [hunk 1]
```diff
@@ -1,312 +1,326 @@
...
+            collateral: {
+                collateralAmount: 10,
+                collateralTokenSymbol: "MKR",
+                gracePeriodInDays: 3,
+            },
+        };
+
+        const collateralizedLoanOrder = {
+            principalTokenSymbol: formData.loan.principalTokenSymbol,
+            principalAmount: new BigNumber(formData.loan.principalAmount * 10 ** 18),
```
Comment (chrismin) [inline __test__/unit/modules/RequestLoan/RequestLoanForm/RequestLoanForm.test.tsx]: same BigNumber comment as before, for all the BigNumbers in this object

File: __test__/unit/modules/RequestLoan/RequestLoanForm/RequestLoanForm.test.tsx [hunk 2]
```diff
@@ -1,312 +1,326 @@
...
+        });
+
+        it("should set interestRate", () => {
+            const spy = jest.spyOn(RequestLoanForm.prototype, "setState");
+            const wrapper = shallow(<RequestLoanForm {...props} />);
+            const formData = {
+                terms: {
+                    interestRate: new BigNumber(3.2),
+                },
+            };
```
[the remark this answers is not in the document]
Comment (nadavhollander) [reply __test__/unit/modules/RequestLoan/RequestLoanForm/RequestLoanForm.test.tsx]: Agreed w/ @saturnial

File: __test__/unit/modules/RequestLoan/RequestLoanForm/RequestLoanForm.test.tsx (hunk 1 above)
Comment (nadavhollander) [reply __test__/unit/modules/RequestLoan/RequestLoanForm/RequestLoanForm.test.tsx]: Yeah -- please loop me in on these discussions.  I don't think consumers of `dharma.js` should ever have to be concerned about decimal amounts.  For now, I agree w/ Chris' concerns re: BigNumber

File: __test__/unit/modules/RequestLoan/RequestLoanForm/RequestLoanForm.test.tsx [hunk 3]
```diff
@@ -1,312 +1,311 @@
...
+        });
+    });
+
+    describe("#handleSignDebtOrder", () => {
+        let debtOrder;
+        beforeEach(() => {
+            debtOrder =
+                '{"principalToken":"0x9b62bd396837417ce319e2e5c8845a5a960010ea","principalAmount":"10","termsContract":"0x1c907384489d939400fa5c6571d8aad778213d74","termsContractParameters":"0x0000000000000000000000000000008500000000000000000000000000000064","kernelVersion":"0x89c5b853e9e32bf47c7da1ccb02e981b74c47f2f","issuanceVersion":"0x1d8e76d2022e017c6c276b44cb2e4c71bd3cc3de","debtor":"0x431194c3e0f35bc7f1266ec6bb85e0c5ec554935","debtorFee":"0","creditor":"0x0000000000000000000000000000000000000000","creditorFee":"0","relayer":"0x0000000000000000000000000000000000000000","relayerFee":"0","underwriter":"0x0000000000000000000000000000000000000000","underwriterFee":"0","underwriterRiskRating":"0","expirationTimestampInSec":"1524613355","salt":"0","debtorSignature":{"v":1,"r":"sometext","s":"sometext"},"creditorSignature":{"v":27,"r":"0xc5c0aaf7b812cb865aef48958e2d39686a13c292f8bd4a82d7b43d833fb5047d","s":"0x2fbbe9f0b8e12ed2875905740fa010bbe710c3e0c131f1efe14fb41bb7921788"},"underwriterSignature":{"r":"","s":"","v":0}}';
```
Comment (nadavhollander) [inline __test__/unit/modules/RequestLoan/RequestLoanForm/RequestLoanForm.test.tsx]: Is there any reason why this isn't formatted in a more readable manner?

File: src/modules/RequestLoan/RequestLoanForm/RequestLoanForm.tsx [hunk 4]
```diff
@@ -108,32 +105,28 @@ class RequestLoanForm extends React.Component<Props, State> {
...
-                debtOrder = await dharma.adapters.simpleInterestLoan.toDebtOrder(loanOrder);
-            } else {
-                const collateralData = {
-                    collateralAmount: new BigNumber(collateralAmount),
-                    collateralTokenSymbol,
-                    gracePeriodInDays: new BigNumber(gracePeriodInDays),
-                };
-                const collateralizedLoanOrder = Object.assign(loanOrder, collateralData);
+            const collateralData = {
+                collateralAmount: new BigNumber(collateralAmount * 10 ** 18), // scale up user input
```
Comment (nadavhollander) [inline src/modules/RequestLoan/RequestLoanForm/RequestLoanForm.tsx]: Flagging this as Chris' comments re: BigNumber exponentiation need to be addressed here.

File: src/modules/RequestLoan/RequestLoanForm/RequestLoanForm.tsx [hunk 5]
```diff
@@ -193,17 +180,10 @@ class RequestLoanForm extends React.Component<Props, State> {
...
-            if (!collateralized) {
-                generatedDebtOrder = await this.props.dharma.adapters.simpleInterestLoan.fromDebtOrder(
-                    debtOrder,
-                );
-            } else {
-                generatedDebtOrder = await this.props.dharma.adapters.collateralizedSimpleInterestLoan.fromDebtOrder(
-                    debtOrder,
-                );
-            }
+            const generatedDebtOrder = await this.props.dharma.adapters.collateralizedSimpleInterestLoan.fromDebtOrder(
```
Comment (nadavhollander) [inline src/modules/RequestLoan/RequestLoanForm/RequestLoanForm.tsx]: Nit: naming would be more intuitive here if it was `collateralizedLoanOrder` as opposed to `generated_`

File: src/modules/RequestLoan/RequestLoanForm/schema.tsx [hunk 6]
```diff
@@ -57,40 +57,35 @@ export const schema: JSONSchema4 = {
                 termLength: {
                     type: "number",
                     title: "Term Length",
-                    description:
-                        'Enter the length of the entire debt agreement, in units of the chosen installments (e.g. a term length of 2 with an installment type of "monthly" would be equivalent to a 2 month long loan)',
+                    description: singleLineString`Enter the length of the entire debt agreement, in
```
Comment (nadavhollander) [inline src/modules/RequestLoan/RequestLoanForm/schema.tsx]: Nit: please align whole text block for readability, i.e: 

```
description: singleLineString`Enter the length of the entire debt agreement, in
                              units of the chosen installments (e.g. a term length of 2 with an installment
                              type of "monthly" would be equivalent to a 2 month long loan)`
```

File: src/utils/webUtils.ts [hunk 7]
```diff
@@ -42,3 +42,7 @@ export const withCommas = (input: number) => {
 export const displayBalance = (balance: BigNumber, numDecimals: number) => {
     return balance.shift(-numDecimals).toFormat();
 };
+
+export const convertRawNumberToScaledBigNumber = (balance: number, numDecimals: number) => {
```
Comment (graemecode) [inline src/utils/webUtils.ts]: `numberToBigNumber`?

Comment (saturnial) [reply src/utils/webUtils.ts]: How about `numberToScaledBigNumber`?

**Comment by saturnial** [comment]
♻️ @chrismin @nadavhollander @graemecode

**Pull request merged**